src/index.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import kebabCase from 'lodash.kebabcase'
+import reduce from 'lodash.reduce'
+import isString from 'lodash.isstring'
+import isNumber from 'lodash.isnumber'
+import IntlFormat from 'intl-messageformat'
+
+const DEFAULT_LOCALE = 'de_DE'
+
+const CURRENCY_BY_REGION = {
+  AT: 'EUR',
+  CH: 'CHF',
+  DE: 'EUR',
+  FR: 'EUR',
+  GB: 'GPB',
+  US: 'USD'
+}
+
+const passThrough = value => value
+
+const {
+  Provider: TranslationProvider,
+  Consumer: TranslationConsumer
+} = React.createContext(DEFAULT_LOCALE)
+
+const withTranslation = ({ translations, params = {}, mapTranslationsToProps = passThrough, format = {} }) => {
+  const initializedTranslations = initializeTranslations(translations, format)
+
+  const wrapComponent = Component => props => {
+    const translationsForLocale = locale => translateWithDefaults({
+      translations: initializedTranslations,
+      locale: locale || DEFAULT_LOCALE,
+      reducer: templateReducer(templateParamValues(props, params))
+    })
+
+    return (
+      <TranslationConsumer>
+        { locale =>
+          <Component
+            {...props}
+            {...mapTranslationsToProps(translationsForLocale(locale), props)}
+          />
+        }
+      </TranslationConsumer>
+    )
+  }
+
+  return wrapComponent
+}
+
+const moneyFormat = locale => ({
+  number: {
+    money: {
+      currency: CURRENCY_BY_REGION[locale.slice(-2)],
+      minimumFractionDigits: 0,
+      style: 'currency'
+    }
+  }
+})
+
+const applyIntlToTranslations = (translations, locale, format) => (
+  reduce(translations, (result, entry, key) => {
```

Comment:
> The most common use case [for `reduce`] is summing up the values of an array to a single number. `reduce` is indended to be used for aggregation and while it *can* be used for transfomation, it might confuse other programmers.

🤓